### ISSUE TITLE
Cache not working properly, possible probleme after symfony changes

### ISSUE THREAD

**Issue by trmalek** [symfony/symfony#36189]
**Symfony version(s) affected**: 4.4.*

**Description**  
I have problems getting caching to work because all my cache control header look like this:

Cache-Control:max-age=0, must-revalidate, private

**How to reproduce**  
$response->setMaxAge(60);
or
$response->setCache(['max_age' => 60]);

**Additional context**  
I downgraded symfony to 4.3.*, the header are set correctly. (max-age=60)

**Comment by pyguerder** [comment]
I observe a similar behavior on my application. Can this be related to #35496 ?

**Comment by trmalek** [comment]
I agree with you, two different context but the same problem

**Comment by pyguerder** [comment]
I solved it by adding `$response->headers->set(AbstractSessionListener::NO_AUTO_CACHE_CONTROL_HEADER, true);` after setting the custom cache headers.

**Comment by nicolas-grekas** [comment]
If your page uses the session, it's not cacheable. You can force it but you need to be totally sure this is compliant with HTTP caching semantics. You can end up with a cache that displays personal data to users of the public if you blindly call this.

Symfony 5.1 will likely provide a way to know *where* the session is used so that you can figure out what needs to be fixed instead - ie not use it anymore.

TL;DR, no bug here.

**Comment by pyguerder** [comment]
There are different issues here:

1.  I am not using sessions on the pages where I add those headers.
2. The behavior changed from Sf 4.3 to Sf 4.4: cache headers were well defined and tested in my application and it broke when I upgraded to the LTS version...

**Comment by nicolas-grekas** [comment]
> I am not using sessions on the pages where I add those headers.

Maybe you don't, but your app does for you, for some reasons. This is what should be investigated (and what 5.1 will make easier to figure out.)

**Comment by trmalek** [comment]
for me also i am not using sessions on all the pages, but i agree with nicolas. I solved my problem when i disabled session at framework.yaml. But I don't understand what's happen on SF 4.4 ? Why the behavior changed from SF 3.4 to SF 4.4 ?

**Comment by nicolas-grekas** [comment]
That was a bug fix: we now correctly detect when the session is used and we turn the page private as it should be.

Since this is resolved, I'm now closing, thanks for coming here to ask.
